module/zfs/zfs_ioctl.c

```diff
@@ -4888,6 +4906,35 @@ zfs_ioc_userspace_upgrade(zfs_cmd_t *zc)
 	return (error);
 }
 
+/*
+ * inputs:
+ * zc_name		name of filesystem
+ *
+ * outputs:
+ * none
+ */
+static int
+zfs_ioc_userobjspace_upgrade(zfs_cmd_t *zc)
```

Comment:
How about we keep this but convert it to the new ioctl style which uses nvlists and slightly change the behavior.  It could be called `zfs_ioc_upgrade` and be used as a generic entry point to trigger activate features on datasets.  This might be handy if you wanted to activate something but didn't want to mount it.  The input nvlist would just need to contain the dataset name and the features to be activated.

You'd need to extend the `zfs upgrade` command to call this.  And atleast initially I think it would be reasonable to always activate any enabled but not present features.